File: Python/Program/Speaker.py
from Komponen import Komponen

class Speaker(Komponen):
    def __init__(self, daya, frekuensi, merk, nama):
        Komponen.__init__(self, merk, nama)  # Memanggil konstruktor kelas induk
        self.daya = daya
        self.frekuensi = frekuensi

    def get_daya(self):
        return self.daya

    def get_frekuensi(self):
        return self.frekuensi

File: C++/Program/Speaker.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Speaker : public Komponen
{
private:
    int daya; 
    int frekuensi;
public:
    Speaker()
    {

    }

    Speaker(int daya, int frekuensi, string nama, string merk) : Komponen(nama, merk)
    {
        this->daya = daya;
        this->frekuensi = frekuensi;
    }

    void setDaya(int daya)
    {
        this->daya = daya;
    }

    int getDaya()
    {
        return daya;
    }

    void setFrekuensi(int frekuensi)
    {
        this->frekuensi = frekuensi;
    }

    int getFrekuensi()
    {
        return frekuensi;
    }

    ~Speaker() {}
};